src/plugins/spec/spec-new.c

```diff
@@ -0,0 +1,754 @@
+
+	const char * arraySize = keyString (arrayMetaKey);
+	const char * minSize = keyString (arrayMinSizeKey);
+	const char * maxSize = keyString (arrayMaxSizeKey);
+
+	return (arrayMinSizeKey == 0 || elektraStrCmp (minSize, arraySize) < 0) &&
+	       (arrayMaxSizeKey == 0 || elektraStrCmp (maxSize, arraySize) > 0);
+}
+
+/**
+ * Checks if the specification has a collision. A collision is when two specification keys exist, one as wildcard specification,
+ * the other as array specification and it is not clear in this case what the correct specification is.
+ *
+ * Example:
+ * 	spec:/server/_/name => meta:/description = value1
+ * 	spec:/server/#/name => meta:/description = value2
+ *
+ * @param specKeys specification keys to check for collision
+ * @return 0 - if no collision was found
+ * 	   pointer to the key which caused collision
+ */
+static Key * specCollision (KeySet * specKeys)
+{
+	for (elektraCursor it = 0; it < ksGetSize (specKeys); it++)
+	{
+		Key * current = ksAtCursor (specKeys, it);
+
+		if (isWildcardSpecification (current))
+		{
+			const char * wildcardSpec = strchr (keyName (current), '/');
+
+			char * arraySpecKey = elektraMalloc (elektraStrLen (wildcardSpec));
+			replaceCharacter (wildcardSpec, arraySpecKey, '_', '#');
+
+			char * formattedLookupKeyName = elektraFormat ("spec:/%s", arraySpecKey);
+			Key * foundKey = ksLookupByName (specKeys, formattedLookupKeyName, 0);
+			elektraFree (arraySpecKey);
+			elektraFree (formattedLookupKeyName);
+			if (foundKey != 0)
+			{
+				return foundKey;
+			}
+		}
+		else if (isArraySpecification (current))
+		{
+			const char * arraySpec = strchr (keyName (current), '/');
+
+			char * wildcardSpecKey = elektraMalloc (elektraStrLen (arraySpec));
+			replaceCharacter (arraySpec, wildcardSpecKey, '#', '_');
+
+			char * formattedLookupKeyName = elektraFormat ("spec:/%s", wildcardSpecKey);
+			Key * foundKey = ksLookupByName (specKeys, formattedLookupKeyName, 0);
+			elektraFree (wildcardSpecKey);
+			elektraFree (formattedLookupKeyName);
+			if (foundKey != 0)
+			{
+				return foundKey;
+			}
+		}
+	}
+	return 0;
+}
+
+static Key * getMatchingKeyFromKeySet (KeySet * ks, char * name)
+{
+	for (elektraCursor it = 0; it < ksGetSize (ks); it++)
+	{
+		Key * current = ksAtCursor (ks, it);
+		if (elektraStrCmp (strchr (keyName (current), '/'), name) == 0)
+		{
+			return current;
+		}
+	}
+	return 0;
+}
+
+/**
+ * Get the key which contains the array size as meta key.
+ *
+ * @param specKeys the specification keys
+ * @param specKey the specification key to use for lookup
+ * @return 0 - if the key was not found
+ * 	   key - the key which contains the array size
+ */
+static Key * getArraySizeOfArrayParent (KeySet * specKeys, Key * specKey)
+{
+	const char * specKeyName = keyName (specKey);
+	char * copiedKeyName = elektraMalloc (elektraStrLen (specKeyName) + 1);
+	strcpy (copiedKeyName, (char *) specKeyName);
+
+	char * arrayParent = strtok (copiedKeyName, "#");
```

Comment:
Please don't use `strtok`. It is not thread-safe. Use `strtok_r` or `strsep` instead.

Reply:
Thanks!!!!!!